x-pack/plugins/security_solution/public/management/pages/policy/view/artifacts/layout/policy_artifacts_layout.tsx

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useCallback, useState } from 'react';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import {
+  EuiTitle,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiText,
+  EuiSpacer,
+  EuiLink,
+  EuiButton,
+  EuiPageContent,
+} from '@elastic/eui';
+import { useAppUrl } from '../../../../../../common/lib/kibana';
+import { APP_UI_ID } from '../../../../../../../common/constants';
+import { ImmutableObject, PolicyData } from '../../../../../../../common/endpoint/types';
+import { ManagementPageLoader } from '../../../../../components/management_page_loader';
+import { useUrlParams } from '../../../../../components/hooks/use_url_params';
+import { useUserPrivileges } from '../../../../../../common/components/user_privileges';
+import { usePolicyDetailsArtifactsNavigateCallback } from '../../policy_hooks';
+import { ExceptionsListApiClient } from '../../../../../services/exceptions_list/exceptions_list_api_client';
+import { useListArtifact } from '../../../../../hooks/artifacts';
+import { PolicyArtifactsEmptyUnassigned, PolicyArtifactsEmptyUnexisting } from '../empty';
+import { PolicyArtifactsList } from '../list';
+import { PolicyArtifactsFlyout } from '../flyout';
+import { PolicyArtifactsPageLabels, policyArtifactsPageLabels } from '../translations';
+import { PolicyArtifactsDeleteModal } from '../delete_modal';
+import { EventFiltersPageLocation } from '../../../../event_filters/types';
+import { HostIsolationExceptionsPageLocation } from '../../../../host_isolation_exceptions/types';
+import { TrustedAppsListPageLocation } from '../../../../trusted_apps/state';
+
+interface PolicyArtifactsLayoutProps {
+  policyItem?: ImmutableObject<PolicyData> | undefined;
+  /** A list of labels for the given policy artifact page. Not all have to be defined, only those that should override the defaults */
+  labels: PolicyArtifactsPageLabels;
+  getExceptionsListApiClient: () => ExceptionsListApiClient;
+  searcheableFields: string[];
+  getArtifactPath: (
+    location?:
+      | Partial<EventFiltersPageLocation>
+      | Partial<TrustedAppsListPageLocation>
+      | Partial<HostIsolationExceptionsPageLocation>
+  ) => string;
+  getPolicyArtifactsPath: (policyId: string) => string;
+  /** A boolean to check extra privileges for restricted actions, true when it's allowed, false when not */
```

Comment:
What does this do exactly? trying to understand the `restricted actions` comment here

Reply:
Inside the component, there are some areas checking the `canCreateArtifactsByPolicy` privilege. This is generic but, for host isolation exception for example, we need an additional check for `canIsolateHost`. So this is supposed to be an extra privilege that will be added to the generic condition.

Reply:
Ahhh. Ok. This is ok for now, but does tie the component to specific artifact's requirements. Anyway we can push requirements to the input props? Maybe a callback function that can return a `boolean` and then let that callback check the additional requirements are met? that would remove the artifact specific logic out of here.